### PR TITLE
New-world lacking submit failures

### PR DESCRIPTION
**Pull request by cjllanwarne**
Fixed a problem with a test expecting submission failures in the old w…orld, whereas in the new world they would be accepted.

### CODE REVIEW

File: src/main/scala/centaur/test/standard/StandardTestFormat.scala [hunk 1]
```diff
@@ -27,12 +30,7 @@ object StandardTestFormat {
...
 
-  def fromString(testFormat: String): ErrorOr[StandardTestFormat] = {
-    testFormat.toLowerCase match {
-      case "workflowsuccess" => Valid(WorkflowSuccessTest)
-      case "workflowfailure" => Valid(WorkflowFailureTest)
-      case "submissionfailure" => Valid(SubmissionFailureTest)
-      case bad => invalidNel(s"No such test format: $bad")
-    }
-  }
+  def fromString(testFormat: String): ErrorOr[StandardTestFormat] = formats collectFirst {
```
Comment (geoffjentry) [inline src/main/scala/centaur/test/standard/StandardTestFormat.scala]: I find the previous way to be far more clear what's going on

**Comment by geoffjentry** [comment]
Why not remove submission failures if they can no longer exist?

**Comment by cjllanwarne** [comment]
@geoffjentry interesting question. I think they _can_ still exist (e.g. no WDL supplied) but don't currently. What do you want me to do - remove or leave? I can see an argument either way

**Comment by geoffjentry** [comment]
@cjllanwarne Let's chat in person as I interpreted your statements as that it would certainly not exist - so clearly I'm not playing with a full deck at the moment :)

File: src/main/scala/centaur/test/standard/StandardTestFormat.scala [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
-    case SubmissionFailureTest => "fail to submit"
   }
 }
 
 object StandardTestFormat {
-  case object WorkflowSuccessTest extends StandardTestFormat
-  case object WorkflowFailureTest extends StandardTestFormat
-  case object SubmissionFailureTest extends StandardTestFormat
+
+  case object WorkflowSuccessTest extends StandardTestFormat { override val name = "WorkflowSuccess" }
```
Comment (mcovarr) [inline src/main/scala/centaur/test/standard/StandardTestFormat.scala]: How about

``` scala

sealed abstract class StandardTestFormat(val name: String) {
.
.
.
case object WorkflowSuccessTest extends StandardTestFormat("WorkflowSuccess")
```

File: README.md [hunk 3]
```diff
@@ -37,8 +37,7 @@ The `basePath` field is optional, but if supplied all paths will be resolved fro
 
 The `testFormat` field can be one of the following, case insensitive:
 * `workflowsuccess`: The workflow being supplied is expected to successfully complete
-* `workflowfailure`: The workflow being supplied is expected to fail 
-* `submissionfailure`: The workflow being supplied is expected to fail on workflow submission
+* `workflowfailure`: The workflow being supplied is expected to fail
```
Comment (mcovarr) [inline README.md:40]: case inconsistent with the `.test`

**Comment by mcovarr** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/45/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: README.md (hunk 3 above)
Comment (cjllanwarne) [inline README.md:40]: Fixed in the `.test`

File: src/main/scala/centaur/test/standard/StandardTestFormat.scala [hunk 4]
```diff
@@ -28,11 +27,9 @@ object StandardTestFormat {
...
   def fromString(testFormat: String): ErrorOr[StandardTestFormat] = {
-    testFormat.toLowerCase match {
-      case "workflowsuccess" => Valid(WorkflowSuccessTest)
-      case "workflowfailure" => Valid(WorkflowFailureTest)
-      case "submissionfailure" => Valid(SubmissionFailureTest)
-      case bad => invalidNel(s"No such test format: $bad")
-    }
+    val formats = List(WorkflowSuccessTest, WorkflowFailureTest)
+    formats collectFirst {
+      case format if format.name.equalsIgnoreCase(testFormat) => Valid(format)
```
Comment (geoffjentry) [inline src/main/scala/centaur/test/standard/StandardTestFormat.scala:32]: I don't think that this works w/o changing the spec for the .test files (and subsequently all of the .test files) - for instance hello.test has `testFormat: workflowsuccess` and that doesn't match any of your formats. 

One could argue that the format in the `.test` file should have the whole word (I don't really care, it just seemed like less typing) but I still say that the previous scheme was easier to visually parse considering how few cases there were

Comment (cjllanwarne) [inline src/main/scala/centaur/test/standard/StandardTestFormat.scala:32]: Check out `format.name`. ~~I think~~ it takes care of that.
EDIT - I tried it :)

Comment (geoffjentry) [inline src/main/scala/centaur/test/standard/StandardTestFormat.scala:32]: Ah right, I was thinking format was a stringg

**Comment by ruchim** [comment]
👍 LGTM

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/45/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

**Pull request merged**